examples/flutter_gallery/lib/demo/material/slider_demo.dart

```diff
@@ -11,12 +13,124 @@ class SliderDemo extends StatefulWidget {
   _SliderDemoState createState() => new _SliderDemoState();
 }
 
+Path _triangle(double size, Offset thumbCenter, {bool invert: false}) {
+  final Path thumbPath = new Path();
+  final double height = math.sqrt(3.0) / 2.0;
+  final double halfSide = size / 2.0;
+  final double centerHeight = size * height / 3.0;
+  final double sign = invert ? -1.0 : 1.0;
+  thumbPath.moveTo(thumbCenter.dx - halfSide, thumbCenter.dy + sign * centerHeight);
+  thumbPath.lineTo(thumbCenter.dx, thumbCenter.dy - 2.0 * sign * centerHeight);
+  thumbPath.lineTo(thumbCenter.dx + halfSide, thumbCenter.dy + sign * centerHeight);
+  thumbPath.close();
+  return thumbPath;
+}
+
+class _CustomThumbShape extends SliderComponentShape {
+  const _CustomThumbShape();
+  static const double _thumbSize = 4.0;
+  static const double _disabledThumbSize = 4.0;
+
+  @override
+  Size getPreferredSize(bool isEnabled, bool isDiscrete) {
+    return new Size.fromRadius(isEnabled ? _thumbSize : _disabledThumbSize);
+  }
+
+  @override
+  void paint(
+    RenderBox parentBox,
+    PaintingContext context,
+    bool isDiscrete,
+    Offset thumbCenter,
+    Animation<double> activationAnimation,
+    Animation<double> enableAnimation,
+    TextPainter labelPainter,
+    SliderThemeData sliderTheme,
+    TextDirection textDirection,
+    double value,
+  ) {
+    final Canvas canvas = context.canvas;
+    final Tween<double> radiusTween = new Tween<double>(
+      begin: _disabledThumbSize,
+      end: _thumbSize,
+    );
+    final ColorTween colorTween = new ColorTween(
+      begin: sliderTheme.disabledThumbColor,
+      end: sliderTheme.thumbColor,
+    );
```

Comment:
it's really unfortunate that we have to build these objects each time. The way we normally do animations, we pre-prepare the animations with tweens and so on configured when the values change, and then during paint we can very cheaply just use the animation's current value.

Reply:
Yeah, but I didn't want to dictate what the tweens were that they used.  I thought about making these shape objects be more like a RenderObject, where it has setters/getters for each thing, and then can decide to only regenerate things like this when the inputs change (I think I mentioned this to you). Maybe I should try it that way?

Reply:
(I sort of did it that way... I implemented no-op setters on the abstract base, and let them override with their own if they care about it.)